flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/logical/operators.scala

```diff
@@ -235,6 +235,13 @@ case class Aggregate(
     groupingExprs.foreach(validateGroupingExpression)
 
     def validateAggregateExpression(expr: Expression): Unit = expr match {
+      case distinctExpr: DistinctAgg =>
+        distinctExpr.child match {
+          case _: AggFunctionCall =>
+            failValidation("UDAGG should prepend 'distinct' modifier before arguments")
```

Comment:
UDAGG should prepend 'distinct' modifier before arguments, for example, udagg.distinct('a, 'b).
Add an example may be more understandable. :-)